tests/difference/core/model/src/properties.ts

```diff
@@ -182,6 +182,65 @@ function stakingWithoutSlashing(hist: BlockHistory): boolean {
   return true;
 }
 
+/**
+ * Checks the validator set replication property as defined
+ * https://github.com/cosmos/ibc/blob/main/spec/app/ics-028-cross-chain-validation/system_model_and_properties.md#system-properties
+ *
+ * @param hist A history of blocks.
+ * @returns Is the property satisfied?
+ */
+function validatorSetReplication(hist: BlockHistory): boolean {
+  // return true;
+  const blocks = hist.blocks;
+  let good = true;
+  blocks[C].forEach((b: CommittedBlock, hC: number) => {
+    if (hC < 1) {
+      // The model starts at consumer height 0, so there is
+      // no committed block at height - 1. This means it does
+      // not make sense to try to check the property for height 0.
+      return
+    }
+    const ss = b.invariantSnapshot;
+    // Get the vscid of the last update which dictated
+    // the consumer valset valsetC committed at hC-1
+    const vscid = ss.hToVscID[hC];
+    // The VSU packet was sent at height hP-1
+    const hP = ss.vscIDtoH[vscid];
+    // Compare the validator sets at hC-1 and hP-1
+    const valsetC = blocks[C].get(hC - 1)!.invariantSnapshot.consumerPower;
+    // The provider set is implicitly defined by the status and tokens (power)
+    if (hP < 1) {
+      // The model starts at provider height 0, so there is
+      // no committed block at height - 1. This means it does not
+      // make sense to try to check the property for height 0.
+      return
```

Comment:
Nit: I find it less error prone to employ `for (const element of array1)` pattern instead of `object.forEach` if you have cases that would skip over an array element if some condition is met.

`.forEach` is only cleaner when you always process each element in the same way

Reply:
Good point! I've changed it.